### PR TITLE
Add SuperGLUE metric

### PR DESCRIPTION
**Pull request by calpt**
Adds a new metric for the SuperGLUE benchmark (similar to the GLUE benchmark metric).

### CODE REVIEW

File: metrics/super_glue/super_glue.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        f1 = f1_score(y_true=question_labels, y_pred=question_preds)
+        f1s.append(f1)
+        em = int(sum([p == l for p, l in preds_labels]) == len(preds_labels))
+        ems.append(em)
+    avg_f1 = sum(f1s) / len(f1s)
+    avg_em = sum(ems) / len(ems)
+    return {"f1": avg_f1, "exact_match": avg_em}
+
+
+class SuperGlue(datasets.Metric):
```
Comment (lhoestq) [inline metrics/super_glue/super_glue.py:100]:
```suggestion
@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
class SuperGlue(datasets.Metric):
```
Could you also add the kwargs this way please ?

File: metrics/super_glue/super_glue.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        - for 'record': list of question-answers dictionaries with the following keys:
+            - 'idx': index of the question as specified by the dataset
+            - 'answers': list of possible answers
+        - otherwise: list of reference labels
+Returns: depending on the SuperGLUE subset, one or several of:
+    "accuracy": Accuracy
+    "f1": F1 score
+    "exact_match": Exact match between answer and gold answer
+    "matthews_correlation": Matthew Correlation
+"""
```
Comment (lhoestq) [inline metrics/super_glue/super_glue.py:60]: Can you add examples in the docstring please ?
You can get some inspiration from the examples in [glue.py](https://github.com/huggingface/datasets/blob/master/metrics/glue/glue.py):
```

Examples:

    >>> glue_metric = datasets.load_metric('glue', 'sst2')  # 'sst2' or any of ["mnli", "mnli_mismatched", "mnli_matched", "qnli", "rte", "wnli", "hans"]
    >>> references = [0, 1]
    >>> predictions = [0, 1]
    >>> results = glue_metric.compute(predictions=predictions, references=references)
    >>> print(results)
    {'accuracy': 1.0}

    >>> glue_metric = datasets.load_metric('glue', 'mrpc')  # 'mrpc' or 'qqp'
    >>> references = [0, 1]
    >>> predictions = [0, 1]
    >>> results = glue_metric.compute(predictions=predictions, references=references)
    >>> print(results)
    {'accuracy': 1.0, 'f1': 1.0}

    >>> glue_metric = datasets.load_metric('glue', 'stsb')
    >>> references = [0., 1., 2., 3., 4., 5.]
    >>> predictions = [0., 1., 2., 3., 4., 5.]
    >>> results = glue_metric.compute(predictions=predictions, references=references)
    >>> print({"pearson": round(results["pearson"], 2), "spearmanr": round(results["spearmanr"], 2)})
    {'pearson': 1.0, 'spearmanr': 1.0}

    >>> glue_metric = datasets.load_metric('glue', 'cola')
    >>> references = [0, 1]
    >>> predictions = [0, 1]
    >>> results = glue_metric.compute(predictions=predictions, references=references)
    >>> print(results)
    {'matthews_correlation': 1.0}
```

Examples of usage in the docstring are tested using doctest.

**Review by lhoestq — COMMENTED** [review]
Hi ! Sorry for the delay reviewing your Pull Request

Thanks a lot for adding this metric.

Could you add some usage examples in the docstring please ?

File: metrics/super_glue/super_glue.py (hunk 2 above)
Comment (calpt) [reply metrics/super_glue/super_glue.py:106]: Sure, I added examples for the different possible metrics returned.

**Review by lhoestq — APPROVED** [review]
Thanks !

**Pull request merged**